sdk/src/AbstractMain.cpp

```diff
@@ -225,19 +213,25 @@ int AbstractMain::run(int in_argc, char** in_argv)
    if (ret != 0)
       return ret;
 
-   // Remove the stderr log destination.
-   removeLogDestination(stderrLogDest->getId());
-
-   if (options.getLogLevel() > LogLevel::INFO)
-   {
-      addLogDestination(
+   addLogDestination(
          std::unique_ptr<ILogDestination>(
             new FileLogDestination(
-               3,
+               "MainLogDestination",
                options.getLogLevel(),
+               LogMessageFormatType::PRETTY,
                getProgramId(),
-               options.getScratchPath())));
-   }
+               logging::FileLogOptions(options.getLoggingDir())
+               )));
+
+   logging::logInfoMessage("Starting plugin...");
+
+   // Turn on stderr logging while options are parsed.
+   std::shared_ptr<ILogDestination> stderrLogDest(new StderrLogDestination("StderrLogging",LogLevel::INFO, LogMessageFormatType::PRETTY));
```

Comment:
We should also move this addLogDestination for the stderr log destination just after the creation of the file log destination before options are parsed. That way, option parsing errors are presented on the console during startup.

Reply:
Just to confirm. 
1) Set programId().
2) get Options Instance.
3) Add main log destination.
4) Add stderr logDestination.
5) Parse options.
Is it okay to still use the logLevel here? https://github.com/rstudio/rstudio-launcher-plugin-sdk/blob/4c19701cf8f8c2362d755bc0424007d49b31eaff/sdk/src/AbstractMain.cpp#L207

Reply:
Good catch on the getLogLevel issue. We should do the following order:

1
4
2
5
3

This ensures that we have a stderr log destination that will log any errors encountered when parsing options. Then once we know the desired log level, we create the log destination for the file to start logging to file.